doc/sphinx/source/esmvalcore/preprocessor.inc

```diff
@@ -1,36 +1,215 @@
+and they are regularly updated with changing geographical features.
+
+In ESMValTool, land-sea-ice masking can be done in two places: in the
+preprocessor, to apply a mask on the data before any subsequent preprocessing
+step and before running the diagnostic, or in the diagnostic scripts
+themselves. We present both these implementations below.
+
+To mask out a certain domain (e.g., sea) in the preprocessor step,
+`mask_landsea` can be used:
+
+.. code-block:: yaml
+
+    preprocessors:
+      preproc_mask:
+        mask_landsea:
+          mask_out: sea
+
+and requires only one argument:
+* mask_out: either land or sea.
+
+The preprocessor automatically retrieves the corresponding mask (`fx: stfof` in
+this case) and apply it so that sea-covered grid cells are set to
+missing. Conversely, it retrieves the `fx: sftlf` mask when land  need to be
+masked out, respectively. If the corresponding fx file is not found (which is
+the case for some models and almost all observational datasets), the
+preprocessor attempts to mask the data using Natural Earth mask files (that are
+vectorized rasters). As mentioned above, the spatial resolution of the the
+Natural Earth masks are much higher than any typical global model (10m for
+land and 50m for ocean masks).
+
+Note that for masking out ice sheets, the preprocessor uses a different
+function, to ensure that both land and sea or ice can be masked out without
+losing generality. To mask ice out, `mask_landseaice` can be used:
+
+.. code-block:: yaml
+
+  preprocessors:
+    preproc_mask:
+      mask_landseaice:
+        mask_out: ice
+
+and requires only one argument:
+* mask_out: either landsea or ice.
+
+As in the case of `mask_landsea`, the preprocessor automatically retrieves the
+`fx: sftgif` mask.
+
+Another option is to just read the fx masks as any other CMOR variable and use
+it within a diagnostic script. This can be done in the variable dictionary by
+specifiying the desired fx variables (masks):
+
+.. warning::
+  Code snippet, text and link to function to be added (after #1037 and #1075
+  are closed).
+
+
+
+Horizontal regridding
+=====================
+.. warning::
+   Documentation of  _regrid.py (part 2) to be added
+
+
+
+Masking of missing values
+=========================
+.. warning::
+   Documentation of  _mask.py (part 2) to be added
+
+
+Multi-model statistics
+======================
+
+.. warning::
+   Documentation of _multimodel.py to be added.
+
+Information on maximum memory required: In the most general case, we can set upper limits on the maximum memory the anlysis will require:
+
+Ms = (R + N) x F_eff - F_eff - when no multimodel analysis is performed;
+Mm = (2R + N) x F_eff - 2F_eff - when multimodel analysis is performed;
+
+where
+
+Ms: maximum memory for non-multimodel module
+Mm: maximum memory for multimodel module
+R: computational efficiency of module (typically 2-3)
+N: number of datasets
+F_eff: average size of data per dataset where F_eff = e x f x F
```

Comment:
"F" is never defined

Reply:
The multimodel section is to be written yet, this piece of text comes from another section and has just been moved here for now.